Content.Shared/Chemistry/Solution/Solution.cs

```diff
@@ -43,6 +42,47 @@ public class Solution : IEnumerable<Solution.ReagentQuantity>, ISerializationHoo
         /// </summary>
         public Solution() { }
 
+        #region SolutionContainer Fields
+        /// <summary>
+        ///     If reactions will be checked for when adding reagents to the container.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        [DataField("canReact")]
+        public bool CanReact { get; set; } = true;
+
+        /// <summary>
+        ///     Volume needed to fill this container.
+        /// </summary>
+        [ViewVariables]
+        public ReagentUnit EmptyVolume => MaxVolume - CurrentVolume;
+
+        public ReagentUnit RefillSpaceAvailable => EmptyVolume;
+        public ReagentUnit InjectSpaceAvailable => EmptyVolume;
+        public ReagentUnit DrawAvailable => CurrentVolume;
+        public ReagentUnit DrainAvailable => CurrentVolume;
+
+        /// <summary>
+        ///     Checks if a solution can fit into the container.
+        /// </summary>
+        /// <param name="solution">The solution that is trying to be added.</param>
+        /// <returns>If the solution can be fully added.</returns>
+        public bool CanAddSolution(Solution solution)
+        {
+            return solution.TotalVolume <= EmptyVolume;
+        }
+
+        [DataField("maxSpillRefill")] public ReagentUnit MaxSpillRefill { get; set; }
```

Comment:
Same here. These are networked so would be better if they were lightweight.

Reply:
How do I make them lightweight? I asked vera to make a separate class/struct and she insisted I put it on Solution.

Reply:
Elaborate? I don't really know what this is about, and I remember such a thing so being reminded would be nice

Reply:
The talk was about here: https://discord.com/channels/310555209753690112/770682801607278632/872017703085416448

Basically I moved `SolutionContainer` fields/methods to `Solution`